pre_commit/languages/golang.py

```diff
@@ -20,82 +24,139 @@
 from pre_commit.util import rmtree
 
 ENVIRONMENT_DIR = 'golangenv'
-get_default_version = helpers.basic_get_default_version
 health_check = helpers.basic_health_check
 
 
-def get_env_patch(venv: str) -> PatchesT:
+@functools.lru_cache(maxsize=1)
+def get_default_version() -> str:
+    if helpers.exe_exists('go'):
+        return 'system'
+    else:
+        return '1.19.4'  # C.DEFAULT
```

Comment:
this doesn't seem right?

Reply:
I was wondering if:
- I should return directly a default version `X.Y.Z`?
- or I should return `C.DEFAULT` and translate it somewhere in the code to a default version `X.Y.Z`?

Reply:
yeah this should just be `'default'` -- we absolutely cannot have a specific version hardcoded because I do not want to deal with spammy PRs bumping it all the time

Reply:
> yeah this should just be `'default'` -- we absolutely cannot have a specific version hardcoded because I do not want to deal with spammy PRs bumping it all the time

and translate it somewhere in the code to a default version `X.Y.Z`?  (this won't shoo spammy PRs, and I don't think Go provides an alias for stable or latest version)

Reply:
yeah I can't accept that -- you'll have to figure some way to determine the latest available version

Reply:
Looks it good now?